src/Stratis.Bitcoin.Features.MemoryPool/Rules/CreateMempoolEntryMempoolRule.cs

```diff
@@ -0,0 +1,321 @@
+                    if (coins == null)
+                        return false;
+
+                    if (coins.Height == TxMempool.MempoolHeight)
+                    {
+                        // Assume all mempool transaction confirm in the next block
+                        prevheights.Add(tip.Height + 1);
+                    }
+                    else
+                    {
+                        prevheights.Add((int)coins.Height);
+                    }
+                }
+                lockPair = context.Transaction.CalculateSequenceLocks(prevheights.ToArray(), index, flags);
+
+                if (lp != null)
+                {
+                    lp.Height = lockPair.MinHeight;
+                    lp.Time = lockPair.MinTime.ToUnixTimeMilliseconds();
+                    // Also store the hash of the block with the highest height of
+                    // all the blocks which have sequence locked prevouts.
+                    // This hash needs to still be on the chain
+                    // for these LockPoint calculations to be valid
+                    // Note: It is impossible to correctly calculate a maxInputBlock
+                    // if any of the sequence locked inputs depend on unconfirmed txs,
+                    // except in the special case where the relative lock time/height
+                    // is 0, which is equivalent to no sequence lock. Since we assume
+                    // input height of tip+1 for mempool txs and test the resulting
+                    // lockPair from CalculateSequenceLocks against tip+1.  We know
+                    // EvaluateSequenceLocks will fail if there was a non-zero sequence
+                    // lock on a mempool input, so we can use the return value of
+                    // CheckSequenceLocks to indicate the LockPoints validity
+                    int maxInputHeight = 0;
+                    foreach (int height in prevheights)
+                    {
+                        // Can ignore mempool inputs since we'll fail if they had non-zero locks
+                        if (height != tip.Height + 1)
+                        {
+                            maxInputHeight = Math.Max(maxInputHeight, height);
+                        }
+                    }
+
+                    lp.MaxInputBlock = tip.GetAncestor(maxInputHeight);
+                }
+            }
+
+            return lockPair.Evaluate(index);
+        }
+
+        /// <summary>
+        /// Whether transaction inputs are standard.
+        /// Check for standard transaction types.
+        /// </summary>
+        /// <param name="tx">Transaction to verify.</param>
+        /// <param name="mapInputs">Map of previous transactions that have outputs we're spending.</param>
+        /// <returns>Whether all inputs (scriptSigs) use only standard transaction forms.</returns>
+        private bool AreInputsStandard(Network network, Transaction tx, MempoolCoinView mapInputs)
```

Comment:
I wonder if Standardness should be in their own rule (not urgent maybe a future PR)

Reply:
Yeah that makes sense, it can be hard to tell what the extent is of the standardness checks otherwise